core/interception.js

```diff
@@ -454,6 +454,73 @@ NodeHandler.onMessageNodeReceived = async function(currentNode, messageNodes, is
         if (!isAllowed) break;
     }
 
+    if(encNodes[0].viewOnceMessageV2 !== null)
```

Comment:
Let's split this logic into a separate function, in a similar way to how message revoke is handled.